src/js/controller/controller.js

```diff
@@ -577,15 +596,20 @@ Object.assign(Controller.prototype, {
             _actionOnAttach = null;
             checkAutoStartCancelable.cancel();
 
-            _model.set('pauseReason', _getReason(meta));
+            const state = _model.get('state');
+            if (state !== STATE_PAUSED && state !== STATE_IDLE) {
+                const pauseReason = _getReason(meta);
+                _model.set('pauseReason', pauseReason);
+                _model.set('playOnViewable', (pauseReason === 'viewable'));
+            }
```

Comment:
This new code should be a function that only changes the model when we pause ads or the program controller. This way we only update the model before calling one of the pause methods below, and we don't need the new "state" checks you added above (because we're already dealing with state in `switch (state)` and `adState`:

```js
if (adState && adState !== STATE_PAUSED) {
    _updatePauseReason(meta);
    _api.pauseAd(true, meta);
    return;
}
...
case STATE_BUFFERING: {
    _updatePauseReason(meta);
    _programController.pause();
    break;
...

...outside of the _pause function:
function _updatePauseReason(meta) {
    const pauseReason = _getReason(meta);
    _model.set('pauseReason', pauseReason);
    _model.set('playOnViewable', (pauseReason === 'viewable'));
}
```

Reply:
👍